tools/azsdk-cli/docs/specs/3-customizing-client-tsp.spec.md

```diff
@@ -103,44 +164,103 @@ This will live in `eng/common/knowledge/customizing-client-tsp.md` so that all l
 - `--update-commit-sha`/`commitSha`: The SHA of the commit in `azure-rest-api-specs` to use when regenerating the SDK.
-- `--new-gen`(CLI only): Directory for regenerated TypeSpec output (SDK). Optional, defaults to `./tmpgen` within the package path.
 
-**Current behavior**
+#### Current behavior
 
 The tool currently works primarily in an `azure-sdk-for-*` repo.
 
 ```mermaid
 flowchart TD
-    Gen[<b>1. SDK Code Generation</B><br/>• Generate SDK code using updated TypeSpec<br/>• Output to --new-gen directory]
+    Gen[<b>1. SDK Code Generation</B><br/>• Generate SDK code using updated sha<br/>]
 
-    Gen --> Patch[<b>2. Code Customization Updates</b><br/>• AI generates patches from diff + current code<br/>• Apply patches using ClientCustomizationCodePatchTool<br/>• Fallback rules for trivial changes]
+    Gen --> Patch[<b>2. Code Customization Updates</b><br/>• AI generates patches from diff + current code<br/>• Apply patches using ClientCustomizationCodePatchTool<br/>]
 
-    Patch --> Validate[<b>3. Validation and Iteration</b><br/>• Compile SDK code<br/>• If build fails → retry AI patching loop max 2x<br/>• Optional: run tests, surface logical issues]
+    Patch --> Validate[<b>3. Validation and Iteration</b><br/>• Compile SDK code<br/>• If build fails → retry AI patching loop max 2x<br/>]
 
     style Gen fill:#fff9c4
     style Patch fill:#f3e5f5
     style Validate fill:#e8f5e9
 ```
 
-**Proposed Changes**
-TBD
+### Proposed Changes
+
+The existing CustomizedCodeUpdateTool will be enhanced to implement a two-phase customization workflow:
+
+**Enhanced Tool Behavior:**
+
+1. **Phase A – [TypeSpec Customizations](#typespec-customizations):**
+   - Analyze build failures and determine if TypeSpec decorators can resolve issues
+   - Apply `client.tsp` adjustments (decorators, naming, grouping, scope configurations)
+   - Re-run TypeSpec compilation and regenerate SDK code
+   - Validate build and proceed to Phase B only if issues remain
+
+2. **Phase B – [Code Customizations](#code-customizations):**
+   - If Phase A doesn't resolve all issues, apply language-specific code patches
+   - Use existing ClientCustomizationCodePatchTool for SDK code modifications
+   - Apply safe patches: imports, visibility modifiers, reserved keyword renames, annotations
+   - Validate final build and generate consolidated diff
+
+3. **Summary & Approval:**
+   - Generate [consolidated diff](#consolidation-diff) of all changes (spec + SDK code)
+   - Enforce approval before commit (CLI: interactive prompt, MCP: agent UI)
+   - Maximum of 2 fix cycles to prevent infinite loops
+   - Provide next step instructions for users
+
+#### New inputs
+
+- `--package-path`/`packagePath`: The path to the package (SDK) directory to check. Lives in one of the `azure-sdk-for-*` repos.
+- [optional] `--update-commit-sha`/`commitSha`: The SHA of the commit in `azure-rest-api-specs` to use when regenerating the SDK.
```

Comment:
I think we should also include an optional `typespecProjectPath` (or similar name). I've been running mostly from the azure-rest-api-specs repo where I'd want to specify which typespec project I'm working on.